etcd/etcd.go

```diff
@@ -1440,14 +1448,22 @@ func (obj *EmbdEtcd) nominateCallback(re *RE) error {
 			// client connects to one of the obj.endpoints servers...
 			log.Printf("Etcd: Addresses are: %s", addresses)
 
+			surls := obj.serverURLs
+			if len(obj.advertiseServerURLs) > 0 {
+				surls = obj.advertiseServerURLs
+			}
 			// XXX: just put this wherever for now so we don't block
 			// nominate self so "member" list is correct for peers to see
-			Nominate(obj, obj.hostname, obj.serverURLs)
+			Nominate(obj, obj.hostname, surls)
 			// XXX: if this fails, where will we retry this part ?
 		}
 
 		// advertise client urls
-		if curls := obj.clientURLs; len(curls) > 0 {
```

Comment:
you actually don't need to change this first line, it's more concise and idiomatic and happens to do what we want AFAICT.